src/Router.ts

```diff
@@ -318,11 +319,10 @@ export class Router<C extends Context> extends Evented {
 						//
 						// Reset selected routes if not dispatched from start().
 						this._currentSelection = dispatchFromStart ? result : [];
-
-						for (const { handler, params } of result) {
-							catchRejection(this, context, path, handler({ context, params }));
-						}
-
+						const promise = result.reduce((promise, { handler, params }) => {
```

Comment:
The handlers here are actually from each route segment, so the order is important (and the main change of the PR).

An example route setup could be, where the first route needs to resolve before calling the `exec` (or `fallback`) of the child route.

```ts
	const myParentRoute = new Route({
		path: 'route/{id}',
		exec(request) {
			return myAsyncAction(params.id).then((data) => {
				setState(data);
			});
		}
	});

	const myChildRoute = new Route({
		path: 'child/{id}',
		exec(request) {
			// do another action using the data from the parent route
			return myOtherAsyncAction(params.id, state.data).then((data) => {
				setState(data);
			});
		}
	});

	myParentRoute.append(myChildRoute);
```